src/System.Buffers.Primitives/System/Buffers/BufferExtensions.cs

```diff
@@ -15,5 +15,58 @@ public static class BufferExtensions
         {
             return first.Span.SequenceEqual(second.Span);
         }
+
+        public static int LastIndexOf(this Span<byte> buffer, ReadOnlySpan<byte> values)
+        {
+            return LastIndexOf((ReadOnlySpan<byte>)buffer, values);
+        }
+
+        public static int LastIndexOf(this ReadOnlySpan<byte> buffer, ReadOnlySpan<byte> values)
+        {
+            if (buffer.Length < values.Length) return -1;
+            if (values.Length == 0) return 0;
+
+            int candidateLength = buffer.Length;
+            var firstByte = values[0];
+            while (true)
+            {
+                int index = LastIndexOf(buffer.Slice(0, candidateLength), firstByte);
+                if (index == -1) return -1;
+                var slice = buffer.Slice(index);
+                if (slice.StartsWith(values)) return index;
```

Comment:
Should this be values.Slice(1) (along with some bounds checks) since we already found the `firstByte` at `index`?

Reply:
I could have sliced the buffer by index + 1 and then call StartsWith with values.Slice(1), as you said. But, I think all this additional work to slice values and do more arithmetic would be more expensive than searching one byte one more time.

Reply:
That makes sense.